lib/rubocop/cop/mixin/interpolation.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality for working with string interpolations.
+    module Interpolation
+      def on_dstr(node)
+        on_node_with_interpolations(node)
+      end
+
+      alias on_xstr on_dstr
+      alias on_dsym on_dstr
+      alias on_regexp on_dstr
+
+      def on_node_with_interpolations(node)
+        node.each_child_node(:begin) do |begin_node|
+          on_interpolation(begin_node)
+        end
+      end
+
+      def on_interpolation(_begin_node)
```

Comment:
WIP?

Reply:
I think this is a template method that should be defined in subclasses.  It may be clearer to understand if a message is given to an argument.

e.g. `raise NotImplementedError('Method should be overridden in child classes')`

Reply:
To be implemented in classes using that mixin. This was inspired by https://github.com/rubocop-hq/rubocop/blob/16427114b8d80ff15909c68c0f2ac83a3a6271bd/lib/rubocop/cop/mixin/heredoc.rb#L17-L19

If you prefer, I can delete the method.

Reply:
I replaced the method by documentation.